pkg/sql/lease.go

```diff
@@ -111,6 +113,10 @@ func (s *tableVersionState) leaseExpiration() tree.DTimestamp {
 type LeaseStore struct {
 	execCfg *ExecutorConfig
 
+	// group is used for all calls made to acquireNodeLease to prevent
+	// concurrent lease acquisitions from the store.
+	group *singleflight.Group
```

Comment:
This has to be a pointer because all calls to m.LeaseHolder.func()  result in go making a copy of m.LeaseHolder thereby reading the contents of the group and causing the race detector to barf.